samples/Common.c

```diff
@@ -430,6 +443,39 @@ STATUS initializePeerConnection(PSampleConfiguration pSampleConfiguration, PRtcP
     }
 
     CHK_STATUS(createPeerConnection(&configuration, ppRtcPeerConnection));
+
+    if (pSampleConfiguration->useTurn) {
+#ifdef ENABLE_KVS_THREADPOOL
+        AsyncGetIceStruct* pAsyncData = NULL;
+
+        pAsyncData = (AsyncGetIceStruct*) MEMCALLOC(1, SIZEOF(AsyncGetIceStruct));
+        pAsyncData->signalingClientHandle = pSampleConfiguration->signalingClientHandle;
+        pAsyncData->ppRtcPeerConnection = ppRtcPeerConnection;
+        pAsyncData->pUriCount = &(pSampleConfiguration->iceUriCount);
+        CHK_STATUS(peerConnectionAsync(asyncGetIceConfigInfo, (PVOID) pAsyncData));
```

Comment:
I think we should update `pSampleConfiguration->iceUriCount` and then MEMFREE(pAsyncData) in this function instead of within the thread function. We update `pUriCount` there, but then immediately free it, so, `pSampleConfiguration-> iceUriCount ` would not have the right value.

Reply:
`pUriCount` is dereferenced and incremented, which updates `pSampleConfiguration->iceUriCount` 's value. We have to wait to MEMFREE(pAsyncData) until it's actually been used in the thread function, or we could end up accessing already freed memory.